src/schema.ts

```diff
@@ -84,18 +109,42 @@ export class Schema {
     if (encQ.aggregate || encQ.autoCount) {
       return 1;
     } else if (encQ.bin) {
-      return 1; // FIXME
+      // encQ.bin will either be a boolean or a BinQuery
+      var bin: BinQuery;
+      if (typeof encQ.bin === 'boolean') {
+        // autoMaxBins defaults to 10 if channel is EnumSpec
+        bin = {
+          maxbins: autoMaxBins(encQ.channel as Channel)
+        };
+      } else {
+        bin = encQ.bin;
+      }
+      const fieldSchema = this.fieldSchemaIndex[encQ.field as string];
+      if (!fieldSchema.binStats[bin.maxbins as string]) {
```

Comment:
Please declare `const maxbins: any = bin.maxbins` instead of having `as` every single time you use them.